tests/common/test_string_calls.c

```diff
@@ -323,6 +323,127 @@ START_TEST(test_bm2str__overflow_some_bits_undefined)
 }
 END_TEST
 
+START_TEST(test_str2bm__empty_string)
+{
+    int rv;
+    char buff[16];
+
+    static const struct bitmask_string bits[] =
+    {
+        {1 << 0, "BIT_0"},
+        BITMASK_STRING_END_OF_LIST
+    };
+
+    rv = g_str_to_bitmask("", bits, ",", buff, sizeof(buff));
+
+    ck_assert_str_eq(buff, "");
+    ck_assert_int_eq(rv, 0);
+}
+END_TEST
+
+START_TEST(test_str2bm__all_defined)
+{
+    int rv;
+    char buff[16];
+
+    static const struct bitmask_string bits[] =
+    {
+        {1 << 0, "BIT_0"},
+        {1 << 1, "BIT_1"},
+        BITMASK_STRING_END_OF_LIST
+    };
+
+    int bitmask = 1 << 0 | 1 << 1;
+    rv = g_str_to_bitmask("BIT_0,BIT_1", bits, ",", buff, sizeof(buff));
+
+    ck_assert_str_eq(buff, "");
+    ck_assert_int_eq(rv, bitmask);
+}
+END_TEST
+
+START_TEST(test_str2bm__no_defined)
+{
+    int rv;
+    char buff[16];
+
+    static const struct bitmask_string bits[] =
+    {
+        {1 << 0, "BIT_0"},
+        {1 << 1, "BIT_1"},
+        BITMASK_STRING_END_OF_LIST
+    };
+
+    int bitmask = 0;
+    rv = g_str_to_bitmask("BIT_2,BIT_3", bits, ",", buff, sizeof(buff));
+
+    ck_assert_str_eq(buff, "BIT_2,BIT_3");
+    ck_assert_int_eq(rv, bitmask);
+}
+END_TEST
+
+START_TEST(test_str2bm__some_defined)
+{
+    int rv;
+    char buff[16];
+
+    static const struct bitmask_string bits[] =
+    {
+        {1 << 0, "BIT_0"},
+        {1 << 1, "BIT_1"},
+        {1 << 2, "BIT_2"},
+        BITMASK_STRING_END_OF_LIST
+    };
+
+    int bitmask = 1 << 1;
+    rv = g_str_to_bitmask("a,BIT_1,b", bits, ",", buff, sizeof(buff));
+
+    ck_assert_str_eq(buff, "a,b");
+    ck_assert_int_eq(rv, bitmask);
+}
+END_TEST
+
+START_TEST(test_str2bm__overflow_undefined)
+{
+    int rv;
+    char buff[16];
+
+    static const struct bitmask_string bits[] =
+    {
+        {1 << 0, "BIT_0"},
+        {1 << 1, "BIT_1"},
+        {1 << 2, "BIT_2"},
+        BITMASK_STRING_END_OF_LIST
+    };
+
+    int bitmask = 1 << 1;
+    rv = g_str_to_bitmask("123456789,BIT_1,abcdef", bits, ",", buff, sizeof(buff));
+
+    /* abcdef is not filled */
+    ck_assert_str_eq(buff, "123456789,");
+    ck_assert_int_eq(rv, bitmask);
+}
+END_TEST
+
+START_TEST(test_str2bm__delim_slash)
+{
+    int rv;
+    char buff[16];
+
+    static const struct bitmask_string bits[] =
+    {
+        {1 << 0, "BIT_0"},
+        {1 << 1, "BIT_1"},
+        BITMASK_STRING_END_OF_LIST
+    };
+
+    int bitmask = 1 << 0 | 1 << 1;
+    rv = g_str_to_bitmask("BIT_0/BIT_1", bits, "/", buff, sizeof(buff));
+
+    ck_assert_str_eq(buff, "");
+    ck_assert_int_eq(rv, bitmask);
+}
+END_TEST
+
```

Comment:
Suggest here adding a BIT_2 value as well. If you do this, it catches the error in the function where you have a hard-coded "," in the second `strtok()` call.

Reply:
Fixed.